src/chargen.cpp

```diff
@@ -149,7 +149,8 @@ void archetype_selection_parse(struct descriptor_data *d, const char *arg) {
     char helpbuf[100];
     char non_const_arg[MAX_INPUT_LENGTH];
     strncpy(non_const_arg, arg, sizeof(non_const_arg) - 1);
-    char *line = any_one_arg(non_const_arg, helpbuf);
+    char line[MAX_INPUT_LENGTH];
+    strncpy(line, any_one_arg(non_const_arg, helpbuf), MAX_INPUT_LENGTH- 1);

```

Comment:
Since strncpy is not guaranteed to be zero-terminated, I've been going through replacing strncpys with strlcpy wherever I can. Please do the same when you see these!